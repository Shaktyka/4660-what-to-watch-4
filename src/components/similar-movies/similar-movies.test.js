import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import SimilarMovies from './similar-movies.jsx';

const MAX_SIMILAR_FILMS_AMOUNT = 4;

const FILMS_DATA = [{
  bgColor: `#A6B7AC`,
  cover: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
  description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City
    seeking revenge against Bill the Butcher, his father's killer.`,
  director: `Martin Scorsese`,
  duration: 167,
  genre: `Crime`,
  id: 1,
  isFavorite: false,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
  preview: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  ratingCount: 370881,
  ratingScore: 8.8,
  source: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  starring: [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
  title: `Gangs of new york`,
  year: 2002
}];

describe(`SimilarMovies rendering`, () => {

  it(`SimilarMovies renders correctly`, () => {
    const films = FILMS_DATA.slice(0, MAX_SIMILAR_FILMS_AMOUNT);

    const tree = renderer
      .create(
          <BrowserRouter>
            <SimilarMovies
              films={films}
              onFilmCardClick={() => {}}
              onHoverCard={() => {}}
              isLoading={false}
            />
          </BrowserRouter>, {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
