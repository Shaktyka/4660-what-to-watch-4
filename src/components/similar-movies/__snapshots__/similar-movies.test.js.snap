// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimilarMovies rendering SimilarMovies renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    id={1}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/bohemian-rhapsody.jpg"
        type="video/webm"
        width="280"
      >
        Sorry, your browser doesn't support embedded video, but don't worry, you can 
        <a
          href="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        >
          download it
        </a>
        and watch it with your favorite video player!
      </video>
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id={2}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/dardjeeling-limited.jpg"
        type="video/webm"
        width="280"
      >
        Sorry, your browser doesn't support embedded video, but don't worry, you can 
        <a
          href="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        >
          download it
        </a>
        and watch it with your favorite video player!
      </video>
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Dardjeeling Limited
      </a>
    </h3>
  </article>
</div>
`;
